misc/test/examples_test.go

```diff
@@ -223,6 +223,15 @@ func TestAccAwsPyAppSync(t *testing.T) {
 	integration.ProgramTest(t, &test)
 }
 
+func TestAccAwsGoAppSync(t *testing.T) {
+	test := getAWSBase(t).
+		With(integration.ProgramTestOptions{
+			Dir: path.Join(getCwd(t), "..", "..", "aws-go-appsync"),
+		})
```

Comment:
you can also include additional runtime validation here to make this test more robust. You can translate that curl command into an HTTP request, some of the other examples already do this in their tests: https://github.com/pulumi/examples/blob/1c721ea2435478165850f041d14bdfd50c29d2d4/misc/test/examples_test.go#L48